RedHatInsights/insights-advisor-frontend
# Bulk Selector Refactor with loading state

With this PR bulk selector refactor is completed. When pressing the 'Select All' option in the dropdown, it will select all systems associated with the currently set filters. So if you were to filter down to 400 systems (starting with lets say 2,000), when pressing selectAll, it will making paginated calls to the api and select the 400 systems as opposed to the 2,000. This PR also adds a loading state to the bulk Selector, and makes bulkSelector work as UX team intended it too. 

 npm run start:proxy:beta
To test the Select All
- Navigate to advisor recommendations in Beta Env,
- Select a pathway with alot of systems (you can click the systems affected link)
- With the systems tab selected, press select all in the bulk selector and it will select every system and while promise is resolving a spinner will appear - old behavior was it selected 1000

- Then deselect all
- Filter down the total systems (type in 'rhiq' and it should filter out something like 50 systems if testing in Insights-qa account) 
- Press selectAll in bulk Selector
- All filtered down results are selected as opposed to all systems associated with the pathway

Bulk Selector should 
-have 'select page' option, that simply selects every item on the current page if not already selected
-Pressing the square should only deselect or selectAll systems
-if the current page is selected, pressed 'select Page' should not do anything, it is only meant for that 1 action
